### DISCUSSION TITLE
📌RT-Thread 4.1.1 路线与意见征集

### DISCUSSION THREAD

**Discussion by mysterywolf** [RT-Thread/rt-thread#6223] [Ideas]
4.1.0已经发布，接下来将发布4.1.1版本。按照版本发布原则，4.1.1是4.1.0的bug修正版本，其修改幅度不能过于剧烈。4.1.1的计划内容包含：
- 社区在使用4.1.0发布版本时反馈的问题
- 4.1.0征集意见时提出，但是未计划在4.1.0中修改的部分

之前4.1.0路线和意见征集issue参见指路：https://github.com/RT-Thread/rt-thread/issues/5342
5.0.0意见征集指路：https://github.com/RT-Thread/rt-thread/issues/5616

**Comment by jiuxiaxixi** [comment]
请修复通过RTT  STUDIO  创建的项目无法枚举USB的BUG  #5799

**Comment by BernardXiong** [comment]
> 4.1.0已经发布，接下来将发布4.1.1版本。按照版本发布原则，4.1.1是4.1.0的bug修正版本，其修改幅度不能过于剧烈。4.1.1的计划内容包含：
> 
> * 社区在使用4.1.0发布版本时反馈的问题
> * 4.1.0征集意见时提出，但是未计划在4.1.0中修改的部分
> 
> 之前4.1.0路线和意见征集issue参见指路：#5342 5.0.0意见征集指路：#5616

4.1.0发布后，新特性应该都会冻结了的，主要是修bug

**Comment by mysterywolf** [comment]
这个pr之前提交又revert回来了，考虑在4.1.1 重新提交一下 https://github.com/RT-Thread/rt-thread/pull/5742

**Comment by mysterywolf** [comment]
rt_kprintf需要精炼和整理一下，函数里有一些没有用的参数或者代码。比如RT_PRINTF_PRECISION这个宏，必须定义，不定义会直接报错。64位行为优化

**Comment by mysterywolf** [comment]
- CMSIS文件太多重复了
  - https://github.com/RT-Thread/rt-thread/pull/5908  是否对全仓库ARM 的BSP都进行此处理？谁用到了谁按照此处理即可
  - CMSIS重复文件太多 https://github.com/RT-Thread/rt-thread/pull/5998
  - [ ] 大部分BSP都没有选定 ARCH_ARM_CORTEX_FPU 导致CMSIS软件包无法启用FPU相关
- [x] AT32的两份rt_driver可否考虑合并到一起 https://github.com/RT-Thread/rt-thread/pull/5970

**Comment by mysterywolf** [comment]
设备框架胡乱返回RT_EOK错误码的问题：
https://github.com/RT-Thread/rt-thread/pull/5898
http://bbs.eeworld.com.cn/thread-1196113-1-1.html

**Comment by mysterywolf** [comment]
- [x] F469触摸这块初始化逻辑有问题，创建一个线程去初始化另一个线程再去初始化触摸芯片？？
 - [ ]  F407探索者 stm32l4r9-st-eval 也有类似的问题

**Comment by Guozhanxin** [comment]
- [x] free 命令兼容多堆的特性。

https://github.com/RT-Thread/rt-thread/pull/6084

**Comment by Guozhanxin** [comment]
- [x] 完善 vs 上 libc，能编译过 lwip 和sal

**Comment by mysterywolf** [comment]
POSIX
BUG：
- [ ] https://github.com/RT-Thread/rt-thread/blob/master/components/libc/posix/ipc/mqueue.c#L209
- [x] pthread_exit https://github.com/RT-Thread/rt-thread/blob/master/components/libc/compilers/common/stdlib.c#L17

**Comment by mysterywolf** [comment]
BSP:
STM32 pwm驱动在L4下 TIM17可能没有这个定时器：https://club.rt-thread.org/ask/question/0876e296d846a5e1.html

**Comment by dongly** [comment]
free 可否增加实际空余内存显示 free? 每次查看都是要计算一下 free = total - used

**Comment by mysterywolf** [comment]
RTC 还差最后一个问题：https://github.com/RT-Thread/rt-thread/pull/5994#issuecomment-1137399722

**Comment by mysterywolf** [comment]
libc
- [x] string.h中部分glibc/POSIX拓展函数各个编译平台不一致
   例如strnlen 在IAR中定义 但是keil gcc下没有定义,IAR取消扩展C函数
   取消strnlen函数在rtthread.h中的定义
https://github.com/RT-Thread/rt-thread/pull/6026/files

**Comment by mysterywolf** [comment]
## kernel 内核相关
警告消除：https://github.com/RT-Thread/rt-thread/pull/5744

断言和参数检查
https://github.com/RT-Thread/rt-thread/issues/5939
https://github.com/RT-Thread/rt-thread/pull/5850
https://github.com/RT-Thread/rt-thread/pull/5576

**Comment by mysterywolf** [comment]
龙伟：Cortex M 的开关中断保护 用 basepri 这个寄存器
https://www.freertos.org/FreeRTOS_Support_Forum_Archive/February_2014/freertos_Cortex_M3_4_BASEPRI_2cd231d0j.html

**Comment by thewon86** [comment]
#5876

**Comment by mysterywolf** [comment]
发布之前还需要再检查一次的pr
https://github.com/RT-Thread/rt-thread/pull/6011
https://github.com/RT-Thread/rt-thread/pull/6012
https://github.com/RT-Thread/rt-thread/pull/5451

**Comment by mysterywolf** [comment]
https://club.rt-thread.org/ask/question/1dfc5b29df0304a7.html

**Comment by FragrantRye** [comment]
建议修复一下这个问题https://github.com/RT-Thread/rt-thread/issues/6128，

已完成 #6148

**Comment by mysterywolf** [comment]
## 需要继续跟踪的PR和论坛帖子 （转自4.1.0发展规划）
- [ ] #5296 其他系列的STM32是否也有这个问题
- [ ] STM32 LPUART的问题
  - https://club.rt-thread.org/ask/question/433249.html RTT 针对stm32lx系列低功耗串口驱动bug
  - https://club.rt-thread.org/ask/question/424812.html 反馈rtt串口驱动对低功耗串口lpuart1不兼容的问题
  - https://club.rt-thread.org/ask/article/3412.html STM32L4 LPUART1 DMA下不能工作的解决办法
- [x] https://club.rt-thread.org/ask/question/434691.html 论坛反馈的finsh_getchar的一个bug
- [ ] https://club.rt-thread.org/ask/article/54.html STM32 UART
- [ ] https://club.rt-thread.org/ask/question/425502.html STM32L0 drv_usart.c的BUG及解决
- [x] https://club.rt-thread.org/ask/question/429786.html  ulog的问题 https://github.com/RT-Thread/rt-thread/pull/5114
- [ ] https://club.rt-thread.org/ask/question/424912.html drv_pwm.c 有bug
- [x] https://club.rt-thread.org/ask/question/434822.html 同时ulog的timestamp与rtc打印无输出 https://github.com/RT-Thread/rt-thread/pull/5632
- [ ] https://club.rt-thread.org/ask/article/3483.html n32g457 spi驱动
- [ ] https://club.rt-thread.org/ask/article/54.html F030系列芯片UART3~UART6使用问题
- [ ] https://club.rt-thread.org/ask/question/435184.html H7 UART5的问题以及修改方法

**Comment by mysterywolf** [comment]
Scons/ENV
- [ ] https://github.com/RT-Thread/rt-thread/pull/6176#discussion_r927347593
- [ ] https://club.rt-thread.org/ask/article/11d28badc0ddb35b.html
- [ ] 取消自动递归下载submodule，增加pkgs --update-r命令专门负责递归submodule

**Comment by mysterywolf** [comment]
https://github.com/RT-Thread/packages/issues/1287 sal impl 文件夹分散出去
